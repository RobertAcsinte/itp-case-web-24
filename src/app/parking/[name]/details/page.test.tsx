import { afterAll, afterEach, beforeAll, expect, it, } from 'vitest';
import { render } from "@testing-library/react";
import ParkingDetails from './page';
import { http, HttpResponse } from 'msw';
import { setupServer } from 'msw/node';


const parkingPlaces = [
    {
        results: [
            {
                "name": "Test",
                "totalcapacity": 2490,
                "availablecapacity": 2232,
                "occupation": 10,
                "isopennow": 1,
                "categorie": "parking buiten LEZ",
                "description": "Bovengrondse parkeergarage The Loop in Gent",
                "location": {
                  "lon": 3.6864592214319964,
                  "lat": 51.02542334883013
                }
              },
        ]
    },
]

  
export const restHandlers = [
    http.get('https://data.stad.gent/api/explore/v2.1/catalog/datasets/bezetting-parkeergarages-real-time/records?select=name%2C%20totalcapacity%2C%20availablecapacity%2C%20occupation%2C%20isopennow%2C%20categorie%2C%20description%2C%20location&where=name%3D%22Test%22', () => {
        return HttpResponse.json(parkingPlaces[0])
    })
]

const server = setupServer(...restHandlers)

beforeAll(() => server.listen())

afterAll(() => server.close())

afterEach(() => server.resetHandlers())

it('Snapshot Parking Details Page', async () => {
    const searchParams = { params: { name: "Test" } };
    const jsx = await ParkingDetails(searchParams);
    const result = render(jsx)
    expect(result).toMatchSnapshot()
})
