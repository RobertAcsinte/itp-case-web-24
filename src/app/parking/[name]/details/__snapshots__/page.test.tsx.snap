// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot Parking Details Page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="wrapper"
      >
        <div
          class="navbar-title"
        >
          <a
            aria-label="Link to go back to the list of parkings"
            href="/parking"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-left "
              data-icon="arrow-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                fill="currentColor"
              />
            </svg>
          </a>
          <h1
            id="parking-name"
          >
            Test
          </h1>
        </div>
        <section
          aria-labelledby="parking-name"
        >
          <div
            class="card"
          >
            <h2>
              Parking Information
            </h2>
            <p>
              <b>
                Category:
              </b>
               
              parking buiten LEZ
            </p>
            <p>
              <b>
                Description:
              </b>
               
              Bovengrondse parkeergarage The Loop in Gent
            </p>
            <p>
              <b>
                Available spots:
              </b>
               
              2232
            </p>
            <p>
              <b>
                Percentage of taken spots from total spots:
              </b>
               
              0.40
              %
            </p>
            <a
              aria-label="Link to go to Google Maps"
              class="details-link"
              href="https://www.google.com/maps/search/?api=1&query=51.02542334883013%2C3.6864592214319964"
              rel="noopener noreferrer"
              target="_blank"
            >
              Open in Google Maps
            </a>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="wrapper"
    >
      <div
        class="navbar-title"
      >
        <a
          aria-label="Link to go back to the list of parkings"
          href="/parking"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-left "
            data-icon="arrow-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              fill="currentColor"
            />
          </svg>
        </a>
        <h1
          id="parking-name"
        >
          Test
        </h1>
      </div>
      <section
        aria-labelledby="parking-name"
      >
        <div
          class="card"
        >
          <h2>
            Parking Information
          </h2>
          <p>
            <b>
              Category:
            </b>
             
            parking buiten LEZ
          </p>
          <p>
            <b>
              Description:
            </b>
             
            Bovengrondse parkeergarage The Loop in Gent
          </p>
          <p>
            <b>
              Available spots:
            </b>
             
            2232
          </p>
          <p>
            <b>
              Percentage of taken spots from total spots:
            </b>
             
            0.40
            %
          </p>
          <a
            aria-label="Link to go to Google Maps"
            class="details-link"
            href="https://www.google.com/maps/search/?api=1&query=51.02542334883013%2C3.6864592214319964"
            rel="noopener noreferrer"
            target="_blank"
          >
            Open in Google Maps
          </a>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
